src/main/java/org/sonarsource/scanner/maven/bootstrap/JavaVersionResolver.java

```diff
@@ -0,0 +1,114 @@
+
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.lifecycle.LifecycleExecutor;
+import org.apache.maven.plugin.BuildPluginManager;
+import org.apache.maven.plugin.MojoExecution;
+import org.apache.maven.plugin.PluginParameterExpressionEvaluator;
+import org.apache.maven.plugin.descriptor.MojoDescriptor;
+import org.apache.maven.plugin.descriptor.PluginDescriptor;
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.classworlds.realm.ClassRealm;
+import org.codehaus.plexus.component.configurator.converters.ConfigurationConverter;
+import org.codehaus.plexus.component.configurator.converters.basic.StringConverter;
+import org.codehaus.plexus.component.configurator.converters.lookup.ConverterLookup;
+import org.codehaus.plexus.component.configurator.converters.lookup.DefaultConverterLookup;
+import org.codehaus.plexus.configuration.PlexusConfiguration;
+import org.codehaus.plexus.configuration.xml.XmlPlexusConfiguration;
+import org.codehaus.plexus.util.xml.Xpp3Dom;
+
+public class JavaVersionResolver {
+  private static final String MAVEN_COMPILER_PLUGIN = "maven-compiler-plugin";
+  private static final String COMPILE_GOAL = MavenUtils.GROUP_ID_APACHE_MAVEN + ":" + MAVEN_COMPILER_PLUGIN + ":compile";
+
+  private final MavenSession session;
+  private final LifecycleExecutor lifecycleExecutor;
+  private final BuildPluginManager buildPluginManager;
+  private final Log log;
+
+  public JavaVersionResolver(MavenSession session, LifecycleExecutor lifecycleExecutor, BuildPluginManager buildPluginManager, Log log) {
+    this.session = session;
+    this.lifecycleExecutor = lifecycleExecutor;
+    this.buildPluginManager = buildPluginManager;
+    this.log = log;
+  }
+
+  /**
+   * Returns the version of Java used by the maven compiler plugin
+   *
+   * @param pom the project pom
+   * @return the java version
+   */
+  @CheckForNull
+  public String getTarget(MavenProject pom) {
+    return coalesce(getString(COMPILE_GOAL, "target"), MavenUtils.getPluginSetting(pom, MavenUtils.GROUP_ID_APACHE_MAVEN, MAVEN_COMPILER_PLUGIN, "target", null));
+  }
+
+  @CheckForNull
+  public String getSource(MavenProject pom) {
+    return coalesce(getString(COMPILE_GOAL, "source"), MavenUtils.getPluginSetting(pom, MavenUtils.GROUP_ID_APACHE_MAVEN, MAVEN_COMPILER_PLUGIN, "source", null));
+
+  }
+
+  private <T> T coalesce(T... objs) {
+    for (T o : objs) {
+      if (o != null) {
+        return o;
+      }
+    }
+    return null;
+  }
+
+  @CheckForNull
+  private String getString(String goal, String parameter) {
+    try {
+      List<MojoExecution> mojoExecutions = lifecycleExecutor.calculateExecutionPlan(session, true, goal).getMojoExecutions();
+      for (MojoExecution exec : mojoExecutions) {
+        Xpp3Dom configuration = exec.getConfiguration();
+        PlexusConfiguration pomConfiguration = new XmlPlexusConfiguration(configuration);
```

Comment:
Have you tried String version = mojoExecution.getConfiguration().getChildren(parameter)[0].getValue() ?